api/v1/controllers/collectors.js

```diff
@@ -458,14 +489,28 @@ function stopCollector(req, res, next) {
   req.swagger.params.queryBody = {
     value: { status: 'Stopped' },
   };
-  doPatch(req, res, next, helper);
+
+  const resultObj = { reqStartTime: req.timestamp };
+  const requestBody = req.swagger.params.queryBody.value;
+
+  u.findByKey(helper, req.swagger.params)
+  .then((o) => u.isWritable(req, o))
+  .then((o) => {
+    u.patchJsonArrayFields(o, requestBody, helper);
+    u.patchArrayFields(o, requestBody, helper);
+
+    return o.update(requestBody);
+  })
+  .then((retVal) => retVal.reassignGenerators()
+    .then(() => u.handleUpdatePromise(resultObj, req, retVal, helper, res)))
```

Comment:
does this promise need to be nested or can we unwind the nesting?

Reply:
We can unwind it at the cost of storing the results in a variable. Will update it if thats preferable.